File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int low = 0, high = nums.size() - 1, n = nums.size();
        vector<int> res(n);
        for (int i = n - 1; i >= 0; i--) {
            if (nums[low] * nums[low] >= nums[high] * nums[high]) {
                res[i] = nums[low] * nums[low];
                low++;
            } else {
                res[i] = nums[high] * nums[high];
                high--;
            }
        }
        return res;
    }
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int i = 0, j = nums.length - 1, iter = nums.length - 1;
        int[] result = new int[nums.length];
        while (i <= j) {
            int left = nums[i] * nums[i];
            int right = nums[j] * nums[j];
            if (left >= right) {
                result[iter] = left;
                i++;
            } else {
                result[iter] = right;
                j--;
            }
            iter--;
        }
        return result;
    }
}